templates/nft-template/move/sources/nft_launchpad.move

```diff
@@ -0,0 +1,466 @@
+    public fun get_admin(): address acquires Config {
+        let config = borrow_global<Config>(@launchpad_addr);
+        config.admin_addr
+    }
+
+    #[view]
+    public fun get_mint_fee_collector(): address acquires Config {
+        let config = borrow_global<Config>(@launchpad_addr);
+        config.mint_fee_collector_addr
+    }
+
+    #[view]
+    public fun get_registry(): vector<object::Object<collection::Collection>> acquires Registry {
+        let registry = borrow_global<Registry>(@launchpad_addr);
+        registry.collection_objects
+    }
+
+    #[view]
+    public fun get_mint_fee(
+        collection_obj: object::Object<collection::Collection>,
+        stage: string::String,
+    ): u64 acquires CollectionConfig {
+        let collection_config = borrow_global<CollectionConfig>(object::object_address(&collection_obj));
+        let fee = *simple_map::borrow(&collection_config.mint_fee_per_nft_by_stages, &stage);
+        fee
+    }
+
+    // ================================= Helpers ================================= //
+
+    fun is_admin(config: &Config, sender: address): bool {
+        if (sender == config.admin_addr) {
+            true
+        } else {
+            if (object::is_object(@launchpad_addr)) {
+                let obj = object::address_to_object<object::ObjectCore>(@launchpad_addr);
+                object::is_owner(obj, sender)
+            } else {
+                false
+            }
+        }
+    }
+
+    fun is_pending_admin(config: &Config, sender: address): bool {
+        config.pending_admin_addr == option::some(sender)
+    }
+
+    fun pay_for_mint(sender: &signer, mint_fee: u64) acquires Config {
+        if (mint_fee > 0) {
+            aptos_account::transfer(sender, get_mint_fee_collector(), mint_fee)
+        }
+    }
+
+    fun royalty(
+        royalty_numerator: &mut option::Option<u64>,
+        admin_addr: address,
+    ): option::Option<royalty::Royalty> {
+        if (option::is_some(royalty_numerator)) {
+            let num = option::extract(royalty_numerator);
+            option::some(royalty::create(num, 100, admin_addr))
+        } else {
+            option::none()
+        }
+    }
+
+    fun mint_nft_internal(
+        sender_addr: address,
+        collection_obj: object::Object<collection::Collection>,
+        mint_fee: u64,
+    ) acquires CollectionConfig, CollectionOwnerObjConfig {
+        let collection_config = borrow_global<CollectionConfig>(object::object_address(&collection_obj));
+
+        let collection_owner_obj = collection_config.collection_owner_obj;
+        let collection_owner_config = borrow_global<CollectionOwnerObjConfig>(
+            object::object_address(&collection_owner_obj)
+        );
+        let collection_owner_obj_signer = &object::generate_signer_for_extending(&collection_owner_config.extend_ref);
+
+        let next_nft_id = *option::borrow(&collection::count(collection_obj));
+        let nft_obj_constructor_ref = &token::create(
```

Comment:
@BriungRi how do you feel about this part? I want to avoid creator pre mint (because we want to make creator only need to send 1 tx) and forbid users pass in random name.

Reply:
I think the current approach is fine. Later we will add more modules which abstract the minting process, i.e. verify mint stages, payments are made and then mint the relevant token/NFT.